File: 0557 reverse-words-in-a-string-iii/solution.java
class Solution {
    public String reverseWords(String s) {
        int l =0;
        char[] st = s.toCharArray();
        for(int i=0;i<st.length;i++) {
            if(st[i]==' ') {
                rev(st,l,i-1);
                l=i+1;
            }
        }
        rev(st,l,st.length-1);
        return new String(st);
    }
    private void rev(char[] st,int l,int r) {
        while(l<r) {
            char c = st[l];
            st[l++] = st[r];
            st[r--] = c;
        }
    }
}

File: 0557 reverse-words-in-a-string-iii/solution.cpp
class Solution {
public:
    string reverseWords(string s) {
        int l = 0;
        for(int i=0;i<s.size();i++) {
            if(s[i]==' ') {
                rev(s,l,i-1);
                l=i+1;
            }
        }
        rev(s,l,s.size()-1);
        return s;
    }
    void rev(string &s,int l,int r) {
        while(l<r) {
            char c = s[l];
            s[l++] = s[r];
            s[r--] = c;
        }
    }
};
